tests/aws/services/stepfunctions/v2/test_sfn_api.py

```diff
@@ -315,6 +317,102 @@ def test_list_sms(
         lst_resp_filter = [sm for sm in lst_resp["stateMachines"] if sm["name"] in sm_names]
         sfn_snapshot.match("lst_resp_del_filter", lst_resp_filter)
 
+    @markers.aws.validated
+    def test_list_sms_pagination(
+        self, create_iam_role_for_sfn, create_state_machine, sfn_snapshot, aws_client
+    ):
+        snf_role_arn = create_iam_role_for_sfn()
+
+        sfn_snapshot.add_transformer(RegexTransformer(snf_role_arn, "snf_role_arn"))
+
+        definition = BaseTemplate.load_sfn_template(BaseTemplate.BASE_PASS_RESULT)
+        definition_str = json.dumps(definition)
+
+        sm_names = [f"statemachine_{i}_{short_uid()}" for i in range(13)]
+        state_machine_arns = list()
+
+        for i, sm_name in enumerate(sm_names):
+            creation_resp = create_state_machine(
+                name=sm_name,
+                definition=definition_str,
+                roleArn=snf_role_arn,
+            )
+
+            sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_create_arn(creation_resp, i))
+
+            state_machine_arn: str = creation_resp["stateMachineArn"]
+            state_machine_arns.append(state_machine_arn)
+
+        def _list_state_machines(expected_results_count: int, **kwargs):
+            """Returns a filtered list of relevant State Machines"""
+            state_machines = aws_client.stepfunctions.list_state_machines(**kwargs)
+            filtered_sms = [sm for sm in state_machines["stateMachines"] if sm["name"] in sm_names]
```

Comment:
This filtering mitigates flaky tests if other state machines are present. Well done 👏
We often use unique prefixes, but this full match approach works equally well 💡